src/Compilers/Core/Portable/Compilation/ParseOptions.cs

```diff
@@ -137,10 +137,7 @@ protected bool EqualsHelper(ParseOptions other)
             }
 
             return
-                this.SpecifiedKind == other.SpecifiedKind &&
-                this.DocumentationMode == other.DocumentationMode &&
-                this.Features.SequenceEqual(other.Features) &&
-                (this.PreprocessorSymbolNames == null ? other.PreprocessorSymbolNames == null : this.PreprocessorSymbolNames.SequenceEqual(other.PreprocessorSymbolNames, StringComparer.Ordinal));
+                this is { SpecifiedKind: other.SpecifiedKind, DocumentationMode: other.DocumentationMode } && this.Features.SequenceEqual(other.Features) && (this.PreprocessorSymbolNames == null ? other.PreprocessorSymbolNames == null : this.PreprocessorSymbolNames.SequenceEqual(other.PreprocessorSymbolNames, StringComparer.Ordinal));
```

Comment:
>other.SpecifiedKind [](start = 41, length = 19)

This is not a valid pattern.

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344416577